Mathlib/RingTheory/MvPolynomial/Homogeneous.lean

```diff
@@ -545,3 +548,53 @@ theorem decomposition.decompose'_eq :
 end GradedAlgebra
 
 end MvPolynomial
+
+/-- Try to use the universal property of the span (e.g., `Submodule.span_induction`) instead of
+this. -/
+lemma Ideal.exists_isHomogeneous_of_mem_span {ι R : Type*} [CommSemiring R]
+    (x : ι → R) (y : R) (hy : y ∈ Ideal.span (.range x)) :
+    ∃ (p : MvPolynomial ι R), p.IsHomogeneous 1 ∧ p.eval x = y := by
+  induction hy using Submodule.span_induction with
+  | mem s hs =>
+    obtain ⟨i, rfl⟩ := hs
+    exact ⟨MvPolynomial.X i, MvPolynomial.isHomogeneous_X _ _, by simp⟩
+  | zero => exact ⟨0, MvPolynomial.isHomogeneous_zero _ _ _, by simp⟩
+  | add x y _ _ hx hy =>
+    obtain ⟨px, hpxhom, rfl⟩ := hx
+    obtain ⟨py, hpyhom, rfl⟩ := hy
+    exact ⟨px + py, MvPolynomial.IsHomogeneous.add hpxhom hpyhom, by simp⟩
+  | smul a x _ hx =>
+    obtain ⟨px, hpxhom, rfl⟩ := hx
+    exact ⟨MvPolynomial.C a * px, MvPolynomial.IsHomogeneous.C_mul hpxhom a, by simp⟩
+
+/-- Try to use the universal property of the span (e.g., `Submodule.span_induction`) instead of
+this. -/
+lemma Ideal.mem_span_pow_iff {ι R : Type*} [CommSemiring R] {n : ℕ} (x : ι → R) (y : R) :
+    y ∈ (Ideal.span <| Set.range x) ^ n ↔
+      ∃ (p : MvPolynomial ι R), p.IsHomogeneous n ∧ p.eval x = y := by
```

Comment:
The right proof is probably via `homogeneousSubmodule ι R 1 = .span R (Set.range X)` and `homogeneousSubmodule ι R n = homogeneousSubmodule ι R 1 ^ n` and `map_span` and `map_pow`.
And I would prefer having a longer name e.g. `Ideal.mem_span_pow_iff_exists_isHomogeneous`.